oxidation/libparsec/crates/crypto/src/common/sequester.rs

```diff
@@ -11,47 +11,95 @@ pub enum SequesterKeySize {
     _4096Bits = 4096,
 }
 
-pub(crate) trait EnforceSerialize {
-    const ALGORITHM: &'static [u8];
-    fn size_in_bytes(&self) -> usize;
-
-    /// Here we avoid uncessary allocation & enforce output has `key_size`
-    fn serialize(&self, output: &[u8], data: &[u8]) -> Vec<u8> {
-        assert!(output.len() <= self.size_in_bytes());
-        let mut res = vec![0; Self::ALGORITHM.len() + 1 + self.size_in_bytes() + data.len()];
-
-        let (algorithm_part, others) = res.split_at_mut(Self::ALGORITHM.len());
-        let (colon, others) = others.split_at_mut(1);
-        // Here we enforce output has key size with zeros
-        let (_zeros, others) = others.split_at_mut(self.size_in_bytes() - output.len());
-        let (output_part, data_part) = others.split_at_mut(output.len());
-
-        algorithm_part.copy_from_slice(Self::ALGORITHM);
-        colon[0] = b':';
-        output_part.copy_from_slice(output);
-        data_part.copy_from_slice(data);
-
-        res
-    }
-}
+/// Here we avoid uncessary allocation & enforce output has `key_size`
+pub(crate) fn serialize_with_armor(
+    output: &[u8],
+    data: &[u8],
+    key_size_bytes: usize,
+    algo: &str,
+) -> Vec<u8> {
+    // It is import `secret_key_encrypted` has a constant size given this
+    // is how it is retrieved during decryption
```

Comment:
Is the comment well placed, I don't understand it in this context :/

Reply:
The `secret_key_encrypted` is the `output`, but some code are merged, that's why I removed the comment